### DISCUSSION TITLE
Public APIs for enthusiats

### DISCUSSION THREAD

**Discussion by shaman007** [komoot/photon#848] [Q&A] [answered]
I really apologise if that's an offtopic.

As "selfhosting" enthusiast and as the Google is dropping the last service I use from them (Timeline on Maps), I'd set  up local thing, Dawarich that uses Photon on Komoot for geodecoding. I am now importing thins from my takeout and there are about a lot points to decode. Obviously I had got throttled heavily (maybe banned by the API) and I totally undestand that. 

The option for me is to boot up my own OSM infra which it totally fine, but maybe there are asome public API servers with capacity to share or buy. 

I am doing all that thing purelly out of interest, no commertial use, etc is intendent, just looking around how stuff works.

**Comment by lonvia** [comment, marked as answer]
If you got throttled/banned on the komoot server, it is unlikely that you find a public free service that caters for your amount of usage. There are a couple of commercial providers listed at https://wiki.openstreetmap.org/wiki/Nominatim#Alternatives_.2F_Third-party_providers you could ask.

You might want to try rolling your own though. Setting up your own Photon server is meant to be dead simple. Download and extract the database from https://download1.graphhopper.com/public/. Download the photon jar and start it. Enjoy.

**Reply by shaman007** [reply]
Thank you! One more thing if I may ask, the photon jar, does it need separate elasticsearch + postgres? If yes, what RAM/HDD I need?

**Reply by lonvia** [reply]
You can use a separate elasticsearch instance but yo don't need to. Photon comes with elasticsearch directly built in.

For the planet you need 200GB of disks right now (expect this to grow by 10-20GB per year), preferably SSD/NVME. I think you can run it with as little as 16GB RAM but it won't be the fastest. You want at least 64GB, photon.komoot.io runs at 128GB.

**Reply by shaman007** [reply]
Thanks, that mean my main box would serve this well.
